uktrade/data-hub-frontend
# Feature/cls2 152 reuse interaction collection for company activities tab v2

## Description of change

Make it easier to add filters to Company > Activity overview. Started replacing `CompanyActivityFeed` with generic `FilteredCollectionList`.
Implemented basic functionality allowing to filter My interactions 

## Test instructions

Find a company and view it's activities. You should be able to filter on your own interactions and if the company has related companies you should be able to see activities across all related.

On dev database you can try:
[One List Corp](http://localhost:3000/companies/375094ac-f79a-43e5-9c88-059a7caa17f0/activity) and [John Doe](http://localhost:3000/companies/8957ba29-7c84-4493-a5e9-e695c89e8323/activity) have different types of activity (Data Hub and External)
[Alphabet Inc](http://localhost:3000/companies/9aacdbb0-32b3-4dc5-a961-24e6e6c57a46/activity) has related companies and should allow for filtering.

_What should I see?_

## Screenshots

### Before

![image](https://github.com/uktrade/data-hub-frontend/assets/699259/3c6d800c-93eb-4949-93f1-0fbcdb306275)

With related companies
<img width="995" alt="image" src="https://github.com/uktrade/data-hub-frontend/assets/699259/74582c47-ab14-4569-9493-9be60e284bdd">

### After

![image](https://github.com/uktrade/data-hub-frontend/assets/699259/36bda027-a7b1-438e-a400-2971490225ef)

With related companies
![image](https://github.com/uktrade/data-hub-frontend/assets/699259/045ff9c2-354e-4a3d-91e7-3d5e88673e01)


## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
